### PR TITLE
Update 25112023

### PR DESCRIPTION
**Pull request by AnhQuoc2906**
- Add 5 new GET APIs in Character & User API
- Rename services & endpoints

### CODE REVIEW

File: Api/Controllers/CharactersController.cs [hunk 1]
```diff
@@ -38,6 +44,20 @@ public async Task<IActionResult> GetCharacter(Guid id)
         return Ok(character);
     }
 
+    [HttpGet("{id}/character-assets")]
+    public async Task<IActionResult> GetCharAssByCharID(Guid id)
```
Comment (danielNg91) [inline Api/Controllers/CharactersController.cs:48]: Giving the method full name will be better for readability, how the different between **Ass**ignment and **Ass**et? The method should be ListCharAssetsByCharId

File: Application/Business/GameUserServices/GameUserServices.cs [hunk 2]
```diff
@@ -28,8 +28,13 @@ public async Task<ICollection<GameUserEntity>> List()
     }
     public async Task<List<UserEntity>> ListUsersByGameId(Guid id)
     {
-        var game = await _gameUserRepo.WhereAsync(gu => gu.GameId == id,new string[] {"User"});
-        return game.Select(g => g.User).ToList();
+        var user = await _gameUserRepo.WhereAsync(gu => gu.GameId == id,new string[] {"User"});
+        return user.Select(g => g.User).ToList();
+    }
+    public async Task<List<GameEntity>> ListGamesByUserId(Guid id)
```
Comment (danielNg91) [inline Application/Business/GameUserServices/GameUserServices.cs:34]: Do not add a method not to be used anywhere

File: Api/Controllers/CharactersController.cs (hunk 1 above)
Comment (danielNg91) [inline Api/Controllers/CharactersController.cs:48]: Giving the method full name will be better for readability, how the different between Assignment and Asset? The method should be ListCharAssetsByCharId

**Review by danielNg91 — CHANGES_REQUESTED** [review]

File: Application/Business/GameUserServices/GameUserServices.cs (hunk 2 above)
Comment (AnhQuoc2906) [reply Application/Business/GameUserServices/GameUserServices.cs:34]: AnhDP ask me to do it for her this morning though. And this method is use in this API in UserController
   ```
 [HttpGet("{id}/games")]
    public async Task<IActionResult> GetGameByUserID(Guid id)
    {
        return Ok(await _gameUserServices.ListGamesByUserId(id));
    }
```

**Review by danielNg91 — APPROVED** [review]

**Pull request merged**